flake.nix

```diff
@@ -2,61 +2,39 @@
-
-    flake-utils = {
-      url = "github:numtide/flake-utils";
-      inputs.nixpkgs.follows = "nixpkgs";
-    };
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
   };
 
-  outputs = { flake-utils, nixpkgs, self }:
-    flake-utils.lib.eachDefaultSystem
-      (system:
-        let
-          config = {};
-
-          overlays = [
-            # This is an overlay we apply on top of Nixpkgs with some of our
-            # own packages defined.
-            (final: prev: {
-              # A Haskell package set with our own overrides and packages defined.
-              myHaskellPackages = final.haskellPackages.override {
-                overrides = hfinal: hprev: {
-                  # This is our local Haskell package.
-                  example-haskell-nix-flake =
-                    hfinal.callCabal2nix "example-haskell-nix-flake" ./. {};
-                };
-              };
-
-              # This is just a convenient shortcut to our package from the
-              # top-level of Nixpkgs.  We're also applying the
-              # justStaticExecutables function to our package in order to
-              # reduce the size of the output derivation.
-              example-haskell-nix-flake =
-                final.haskell.lib.compose.justStaticExecutables
-                  final.myHaskellPackages.example-haskell-nix-flake;
-
-              # A Haskell development shell for our package that includes
-              # things like cabal and HLS.
-              myDevShell = final.myHaskellPackages.shellFor {
-                packages = p: [ p.example-haskell-nix-flake ];
-
-                nativeBuildInputs = [
-                  final.cabal-install
-                  final.haskellPackages.haskell-language-server
-                ];
-              };
-            })
-          ];
-
-          # Our full Nixpkgs with the above overlay applied.
-          pkgs = import nixpkgs { inherit config overlays system; };
-        in
-        {
-          packages.default = pkgs.example-haskell-nix-flake;
-
-          devShells.default = pkgs.myDevShell;
-        }
-      );
+  outputs = {
+    nixpkgs,
+    self,
+  }: let
+    forEachSystem = f:
+      nixpkgs.lib.genAttrs [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ] (system: f nixpkgs.legacyPackages.${system});
+    package =
+      forEachSystem (pkgs:
+        pkgs.haskellPackages.callCabal2nix "example-haskell-nix-flake" ./. {});
+  in {
+    packages = forEachSystem (pkgs: {
+      default =
+        pkgs.haskell.lib.compose.justStaticExecutables
+        package.${pkgs.system};
+    });
+
+    devShells = forEachSystem (pkgs: {
+      default = pkgs.haskellPackages.shellFor {
+        packages = _: [package.${pkgs.system}];
+
+        nativeBuildInputs = [
+          pkgs.cabal-install
+          pkgs.haskellPackages.haskell-language-server
+        ];
+      };
+    });
```

Comment:
Thanks a lot for fixing this up!

Could you refactor this to make it an overlay (well, a Haskell overlay within a Nixpkgs overlay) on top of Nixpkgs, similar to how the original code is written?

Reply:
it doesn't provide an overlay output so that overlay is pointless and requires instantiating a nixpkgs instance

Reply:
add 
```nix
    overlays.default = final: prev: {
      OverlayedHaskellPackages = final.haskellPackages.override {
        overrides = hfinal: hprev: {
          # This is our local Haskell package.
          example-haskell-package = hfinal.callCabal2nix "example-haskell-nix-flake" ./. {};
        };
      };
    };
    ```
    will give you a usable external overlay

Reply:
I wanted to do that but override the actual Haskell packages and also provide the binary in the overlay, but that's after I come back from vacation so in 3 days